installation/components/setup_hifiberry.sh

```diff
@@ -0,0 +1,119 @@
+    ["hifiberry-dacplusadc"]="HiFiBerry DAC+ ADC"
+    ["hifiberry-dacplusadcpro"]="HiFiBerry DAC+ ADC Pro"
+    ["hifiberry-digi"]="HiFiBerry Digi+"
+    ["hifiberry-digi-pro"]="HiFiBerry Digi+ Pro"
+    ["hifiberry-amp"]="HiFiBerry Amp+ (not Amp2)"
+    ["hifiberry-amp3"]="HiFiBerry Amp3"
+)
+
+example_usage() {
+    for key in "${!hifiberry_map[@]}"; do
+        description="${hifiberry_map[$key]}"
+        echo "$key) $description"
+    done
+    echo "Example usage: ./${script_name} enable hifiberry-dac"
+}
+
+enable_hifiberry() {
+    echo "Enabling HiFiBerry board..."
+    grep -qxF "dtoverlay=$1" "$boot_config_path" || echo "dtoverlay=$1" | sudo tee -a "$boot_config_path" > /dev/null
+    ./../options/onboard_sound.sh disable
+}
+
+disable_hifiberry() {
+    echo "Removing existing HiFiBerry configuration..."
+    sudo sed -i '/dtoverlay=hifiberry-/d' "$boot_config_path"
+    ./../options/onboard_sound.sh enable
+}
+
+check_existing_hifiberry() {
+    existing_config=$(grep 'dtoverlay=hifiberry-' "$boot_config_path")
+    if [ ! -z "$existing_config" ]; then
+        if [ "$1" = "silent" ]; then
+            disable_hifiberry
+            return 0
+        fi
+
+        echo "Existing HiFiBerry configuration detected: $existing_config"
+        read -p "Do you want to proceed with a new configuration? This will remove the existing one. (Y/n): " choice
+        case $choice in
+            [nN][oO]|[nN])
+                echo "Exiting without making changes.";
+                exit;;
+            *)
+                disable_hifiberry;
+                return 0;;
+        esac
+    fi
+}
+
+# 1-line installation
+if [ $# -ge 1 ]; then
+    if [[ "$1" != "enable" && "$1" != "disable" ]] || [[ "$1" == "enable" && -z "$2" ]]; then
+        echo "Error: Invalid arguments provided.
+Usage: ./${script_name} <status> <hifiberry-board>
+where <status> can be 'enable' or 'disable'.
+
+The following board options exist:"
+        example_usage
+        exit 1
+    fi
+
+    if [ "$1" == "enable" ]; then
+        if [[ -v hifiberry_map["$2"] ]]; then
+            check_existing_hifiberry "silent"
+            enable_hifiberry "$2"
+            exit 1
+        fi
+
+        echo "'$2' is not a valid option. You can choose from:"
+        example_usage
+        exit 1
+    fi
+
+    disable_hifiberry
+    exit 1
+fi
+
+# Guided installation
+board_count=${#hifiberry_map[@]}
+counter=1
+
+echo "Select your HiFiBerry board:"
+for key in "${!hifiberry_map[@]}"; do
+    description="${hifiberry_map[$key]}"
+    echo "$counter) $description"
+    ((counter++))
+done
+echo "0) Remove existing HiFiBerry configuration"
+
+read -p "Enter your choice (0-$board_count): " choice
+case $choice in
+    [0])
+        disable_hifiberry;
+        ;;
+    [1-$board_count])
+        check_existing_hifiberry
+        selected_board=$(get_key_by_item_number hifiberry_map "$choice")
+        enable_hifiberry "$selected_board";
```

Comment:
Read out `choice` needs to be the first step. Otherwise its not set after `check_existing_hifiberry` does perform some actions.
```suggestion
        selected_board=$(get_key_by_item_number hifiberry_map "$choice")
        check_existing_hifiberry
        enable_hifiberry "$selected_board";
```

Reply:
It works for me quite well, tbh!

Reply:
On the second run when the previous configuration will be removed, I get errors in line 13 of the helper script like " not an integer".
Debugged and "choice" wasn't set.

But I can recheck later

Reply:
Could not reproduce it, but I will change it as it does not make much difference